Run recovery mechanism doesn't work in all cases

Hello @kinow 

The work on the ssh connectivy issues doesn't seem to be enough

There is a new type of issue coming from the Marenostrum5 that I didn't see before: 


```
2026-02-20 18:25:40,851 Executing command mkdir -p /gpfs/scratch/ehpc01/bsc032056/a315/LOG_a315, retry #1 out of 3
[ERROR] 2026-02-20 18:25:41,008 Command mkdir -p /gpfs/scratch/ehpc01/bsc032056/a315/LOG_a315 in mn5-cluster1 warning: Could not chdir to home directory /home/bsc/bsc032056: Stale file handle
bash: /home/bsc/bsc032056/.bashrc: Stale file handle
mkdir: cannot create directory '/gpfs/scratch': Stale file handle
[eCode=6006]
2026-02-20 18:25:41,008 /gpfs/scratch/ehpc01/bsc032056/a315/LOG_a315 has been created on mn5-cluster1.
[WARNING] 2026-02-20 18:25:41,397 Yaml file /appl/AS/AUTOSUBMIT_DATA/a315/proj/git_project/conf/defaults/defaults_apps.yml not found
[WARNING] 2026-02-20 18:25:41,493 Yaml file /appl/AS/AUTOSUBMIT_DATA/a315/proj/git_project/conf/additional_jobs/wildfires_fwi-False.yml not found
[WARNING] 2026-02-20 18:25:41,496 Yaml file /appl/AS/AUTOSUBMIT_DATA/a315/proj/git_project/conf/additional_jobs/wildfires_wise-False.yml not found
[WARNING] 2026-02-20 18:25:41,532 Yaml file /appl/AS/AUTOSUBMIT_DATA/a315/proj/git_project/conf/additional_jobs/energy_offshore-False.yml not found
[WARNING] 2026-02-20 18:25:41,537 Yaml file /appl/AS/AUTOSUBMIT_DATA/a315/proj/git_project/conf/additional_jobs/obsall-False.yml not found
[WARNING] 2026-02-20 18:25:41,542 Yaml file /appl/AS/AUTOSUBMIT_DATA/a315/proj/git_project/conf/additional_jobs/hydroland-False.yml not found
[WARNING] 2026-02-20 18:25:41,546 Yaml file /appl/AS/AUTOSUBMIT_DATA/a315/proj/git_project/conf/additional_jobs/hydromet-False.yml not found
[ERROR] 2026-02-20 18:25:42,056 Arguments provided: Namespace(version=False, logfile='DEBUG', logconsole='WARNING', command='run', expid='a315', notransitive=False, update_version=False, start_time=None, start_after=None, run_only_members=None, profile=False)
[ERROR] 2026-02-20 18:25:42,057 This is the experiment: <a315> which had an issue with the command: <run> and it is currently using the Autosubmit Version: 4.1.16.
[CRITICAL] 2026-02-20 18:25:42,059 Traceback (most recent call last):
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/autosubmit/platforms/paramiko_platform.py", line 523, in send_file
    self._ftpChannel.put(local_path, remote_path)
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/paramiko/sftp_client.py", line 759, in put
    return self.putfo(fl, remotepath, file_size, callback, confirm)
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/paramiko/sftp_client.py", line 714, in putfo
    with self.file(remotepath, "wb") as fr:
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/paramiko/sftp_client.py", line 372, in open
    t, msg = self._request(CMD_OPEN, filename, imode, attrblock)
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/paramiko/sftp_client.py", line 857, in _request
    return self._read_response(num)
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/paramiko/sftp_client.py", line 909, in _read_response
    self._convert_status(msg)
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/paramiko/sftp_client.py", line 942, in _convert_status
    raise IOError(text)
OSError: Failure

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/autosubmit/job/job_packages.py", line 232, in submit
    self._send_files()
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/autosubmit/job/job_packages.py", line 282, in _send_files
    self.platform.send_file(self._job_scripts[job.name])
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/autosubmit/platforms/paramiko_platform.py", line 527, in send_file
    raise AutosubmitError(f'Cannot send file {local_path} to {remote_path}. '
autosubmit.log.log.AutosubmitError: Cannot send file /appl/AS/AUTOSUBMIT_DATA/a315/tmp/a315_19900101_fc0_186_22_OPA_ENERGYTDIG2.cmd to /gpfs/scratch/ehpc01/bsc032056/a315/LOG_a315/a315_19900101_fc0_186_22_OPA_ENERGYTDIG2.cmd. Connection does not appear to be active: Failure
Error code: 6004

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/autosubmit/autosubmit.py", line 2179, in run_experiment
    Autosubmit.submit_ready_jobs(as_conf, job_list, platforms_to_test, packages_persistence,
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/autosubmit/autosubmit.py", line 2460, in submit_ready_jobs
    platform.submit_ready_jobs(as_conf, job_list, packages_persistence, packages_to_submit,
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/autosubmit/platforms/platform.py", line 426, in submit_ready_jobs
    package.submit(as_conf, job_list.parameters, inspect, hold=hold)
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/autosubmit/job/job_packages.py", line 238, in submit
    raise AutosubmitCritical(f"Error while submitting jobs: {e}", 7013)
autosubmit.log.log.AutosubmitCritical: Error while submitting jobs: Cannot send file /appl/AS/AUTOSUBMIT_DATA/a315/tmp/a315_19900101_fc0_186_22_OPA_ENERGYTDIG2.cmd to /gpfs/scratch/ehpc01/bsc032056/a315/LOG_a315/a315_19900101_fc0_186_22_OPA_ENERGYTDIG2.cmd. Connection does not appear to be active: Failure
Error code: 6004
Error code: 7013

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/autosubmit/scripts/autosubmit.py", line 105, in main
    return_value = Autosubmit.run_command(args)
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/autosubmit/autosubmit.py", line 761, in run_command
    return Autosubmit.run_experiment(args.expid, args.start_time, args.start_after, args.run_only_members,
  File "/appl/AS/4.1.16-dev-6605a3d/lib64/python3.9/site-packages/autosubmit/autosubmit.py", line 2280, in run_experiment
    raise AutosubmitCritical(e.message, e.code, e.trace)
autosubmit.log.log.AutosubmitCritical: Error while submitting jobs: Cannot send file /appl/AS/AUTOSUBMIT_DATA/a315/tmp/a315_19900101_fc0_186_22_OPA_ENERGYTDIG2.cmd to /gpfs/scratch/ehpc01/bsc032056/a315/LOG_a315/a315_19900101_fc0_186_22_OPA_ENERGYTDIG2.cmd. Connection does not appear to be active: Failure
Error code: 6004
Error code: 7013

[CRITICAL] 2026-02-20 18:25:42,059 Error while submitting jobs: Cannot send file /appl/AS/AUTOSUBMIT_DATA/a315/tmp/a315_19900101_fc0_186_22_OPA_ENERGYTDIG2.cmd to /gpfs/scratch/ehpc01/bsc032056/a315/LOG_a315/a315_19900101_fc0_186_22_OPA_ENERGYTDIG2.cmd. Connection does not appear to be active: Failure
Error code: 6004 [eCode=7013]
2026-02-20 18:25:42,059 More info at https://autosubmit.readthedocs.io/en/master/troubleshooting/error-codes.html
```

Result: 

Hard crash of the autosubmit run

Expected outcome: 

Autosubmit recovery mode is triggered -> Connection is renewed -> continue run

--- a/autosubmit/platforms/paramiko_platform.py
+++ b/autosubmit/platforms/paramiko_platform.py
@@ -1213,7 +1213,7 @@ class ParamikoPlatform(Platform):
                 stdout = chan.makefile('rb', bufsize)
                 stderr = chan.makefile_stderr('rb', bufsize)
                 return stdin, stdout, stderr
-            except (paramiko.SSHException, ConnectionError, socket.error) as e:
+            except (paramiko.SSHException, ConnectionError, socket.error, IOError) as e:
                 Log.warning(f'A networking error occurred while executing command [{command}]: {str(e)}')
                 if not self.connected or not self.transport or not self.transport.active:
                     self.restore_connection(None)

--- a/test/integration/test_paramiko_platform.py
+++ b/test/integration/test_paramiko_platform.py
@@ -538,12 +538,14 @@ def test_exec_command_ssh_session_not_active(
     [
         paramiko.ssh_exception.NoValidConnectionsError({'192.168.0.1': ValueError('failed')}),  # type: ignore
         ConnectionError('Someone unplugged the networking cable.'),
-        socket.error('A random socket error occurred!')
+        socket.error('A random socket error occurred!'),
+        IOError('Someone plugged the cable off.')
     ],
     ids=[
         'paramiko ssh exception',
         'connection error',
-        'socket error'
+        'socket error',
+        'io error'
     ]
 )
 @pytest.mark.ssh

